src/sentry/rules/processor.py

```diff
@@ -194,6 +253,12 @@ def apply(self):
             return {}.values()
 
         self.grouped_futures.clear()
-        for rule in self.get_rules():
-            self.apply_rule(rule)
+        rules = self.get_rules()
+        if len(rules) == 1:
+            rule = rules[0]
+            rule_statuses = {rule.id: self.get_rule_status(rule)}
```

Comment:
I chose to do this because it results in fewer queries for single rule projects. Not sure if it's really worth having the extra logic around to handle the case, so if anyone feels strongly I can just combine them.

Reply:
I think this is fine, I would just document it in code as I would imagine might be confusing for us / anyone else to read in the future

Reply:
I started writing the comment and then found it hard to justify, so I just removed this in the end